### DISCUSSION TITLE
server-side possibility?

### DISCUSSION THREAD

**Discussion by enisbu** [TranscribeJs/transcribe.js#2] [General]
First off all, I want to say your work on transcribe.js is truly impressive - kudos to you!

I’m planning to integrate the library into a project and was wondering if there’s a way to adapt it for server-side usage (for API-Endpoint usage). From what I understand, transcribe.js is primarily designed for client-side execution. However, I’d love to know if there’s any potential for running the transcription feature on a server, either through modifications or a different approach.

If you have any suggestions or could point me in the right direction, I’d really appreciate it! :)

**Comment by thurti** [comment]
Thank you. The actual work is done by [whisper.cpp](https://github.com/ggerganov/whisper.cpp) a C++ implementation of whisper. I've only created a JS api around the wasm build. 

You're right, transcribe.js only works in the browser. This is because it uses the web audio api to load the files and do the sample rate convertion. This api is not available in node.js. 

Technically it would be possible to run the wasm build in node, but I would not recommend that. Instead just run a python (or any other) implemenation of whisper directly on the server. This way you can use larger models and use a GPU for computation which will be soooo much faster.

If you need (for whatever reason) a JS implementation, maybe have a look at https://github.com/xenova/whisper-web . I haven't tried it but there is a good chance this will run in node.

**Reply by enisbu** [reply]
Thank you so much for explaining this to me! You've given me a new perspective on my use case!.. I appreciate you also taking the time to share your insights. And thanks again for all the effort you've put into creating this awesome library - much kudos! :D

**Reply by thurti** [reply]
You're welcome. Good luck with your project.
